hardware_interface/include/hardware_interface/system_interface.hpp

```diff
@@ -98,9 +100,56 @@ class SystemInterface : public rclcpp_lifecycle::node_interfaces::LifecycleNodeI
   virtual CallbackReturn on_init(const HardwareInfo & hardware_info)
   {
     info_ = hardware_info;
+    add_state_interface_descriptions();
+    add_command_interface_descriptions();
     return CallbackReturn::SUCCESS;
   };
 
+  virtual void add_state_interface_descriptions()
+  {
+    for (const auto & joint : info_.joints)
+    {
+      for (const auto & state_interface : joint.state_interfaces)
+      {
+        joint_states_descr_.emplace_back(InterfaceDescription(joint.name, state_interface));
```

Comment:
Just write the full name, it is clearer.
```suggestion
        joint_states_description_.emplace_back(InterfaceDescription(joint.name, state_interface));
```